buildSrc/src/main/java/com/google/firebase/gradle/plugins/PomValidator.kt

```diff
@@ -16,63 +16,77 @@ package com.google.firebase.gradle.plugins
-import org.w3c.dom.Node
-import org.w3c.dom.NodeList
 
+/**
+ * Ensures that pom dependencies are not accidently downgraded.
+ *
+ * Compares the latest pom at gmaven for the given artifact with the one generate for the current
+ * release.
+ *
+ * @throws GradleException if a dependency is found with a degraded version
+ *
+ * @property pomFile The pom file for the current release
+ * @property artifactId The artifactId for the pom parent
+ * @property groupId The groupId for the pom parent
+ */
 abstract class PomValidator : DefaultTask() {
-  @get:InputFile abstract val pomFilePath: Property<File>
+  @get:InputFile abstract val pomFile: Property<File>
   @get:Input abstract val artifactId: Property<String>
   @get:Input abstract val groupId: Property<String>
 
   @TaskAction
   fun run() {
-    val gMavenHelper = GmavenHelper(groupId.get(), artifactId.get())
-    val latestReleasedVersion = gMavenHelper.getLatestReleasedVersion()
-    val releasedVersionPomUrl = gMavenHelper.getPomFileForVersion(latestReleasedVersion)
-    var output: String = diffWithPomFileUrl(releasedVersionPomUrl).trim()
-    if (output.isNotEmpty()) {
-      throw GradleException("${output}\nPlease fix the above errors")
+    var diff = diffWithPomFromURL(getLatestReleasePomUrl())
+
+    if (diff.isNotEmpty()) {
+      throw GradleException("Dependency version errors found:\n${diff}")
     }
   }
 
-  fun getMapFromXml(pomNodeList: NodeList): Map<String, String> {
-    val pomMap = mutableMapOf<String, String>()
-    for (i in 0..pomNodeList.length - 1) {
-      val node: Node = pomNodeList.item(i)
-      if (node.getNodeType() == Node.ELEMENT_NODE) {
-        val element = node as Element
-        val artifact = element.getElementsByTagName("artifactId").item(0).getTextContent()
-        val version = element.getElementsByTagName("version").item(0).getTextContent()
-        pomMap[artifact] = version
-      }
+  private fun getLatestReleasePomUrl() =
+    with(GmavenHelper(groupId.get(), artifactId.get())) {
+      getPomFileForVersion(getLatestReleasedVersion())
     }
-    return pomMap
-  }
 
-  fun diffWithPomFileUrl(pomUrl: String): String {
-    val factory: DocumentBuilderFactory = DocumentBuilderFactory.newInstance()
-    val oldPomBuilder: DocumentBuilder = factory.newDocumentBuilder()
-    val oldPomDoc: Document = oldPomBuilder.parse(URL(pomUrl).openStream())
-    val currentPomBuilder: DocumentBuilder = factory.newDocumentBuilder()
-    val currentPomDoc: Document = currentPomBuilder.parse(pomFilePath.get())
-    val oldPomMap = getMapFromXml(oldPomDoc.getElementsByTagName("dependency"))
-    val currentPomMap = getMapFromXml(currentPomDoc.getElementsByTagName("dependency"))
+  private fun getMapOfDependencies(doc: Element) =
+    doc
+      .findElementsByTag("dependency")
+      .associate { it.textByTag("artifactId") to it.textByTag("version") }
+      .filter { it.key != "javax.inject" } // javax.inject doesn't respect SemVer and doesn't update
```

Comment:
can we have a static list with this? It will be easier to update in the future if more things need to be ignored

Reply:
SGTM, will do!